src/utils/stacUtils.tsx

```diff
@@ -0,0 +1,100 @@
+import { TbAxisX } from "react-icons/tb";
+import { DataPoint } from "@/pages/MapViewer/components/Toolbox/components/ToolboxRow/types";
+import { parseDate, beautifyKey } from "./genericUtils";
+import { Feature } from "typings/stac";
+
+/**
+ * Further discussion needed around this function.
+ * According to the design there are data points under each of the collection titles.
+ * We can't ever assume what these data points are, so we need to be able to handle any data.
+ *
+ * This works as it is, but it's not very pretty.
+ */
+export const parseCollectionDataPoints = (
+  collection: Collection
+): DataPoint[] => {
+  const dataPoints: DataPoint[] = [];
+
+  const { lastUpdated, summaries } = collection;
+  const { temporal } = collection.extent;
+
+  // Add the Last Updated and Temporal Extent data points
+  if (lastUpdated) {
+    dataPoints.push({
+      icon: CiCalendarDate,
+      alt: "Calendar Icon",
+      text: parseDate(lastUpdated),
+      tooltip: "Last Updated",
+    });
+  }
+
+  temporal &&
+    temporal.interval &&
+    dataPoints.push({
+      icon: IoTimeOutline,
+      alt: "Time Icon",
+      text: `${new Date(
+        temporal.interval[0][0]
+      ).toLocaleDateString()} - ${new Date(
+        temporal.interval[0][1]
+      ).toLocaleDateString()}`,
+      tooltip: "Temporal Extent",
+    });
+
+  // Summaries could be anything, so there needs to be some checks and processing.
+  // We may not even want this, but it's here for now.
+  // At least it gives some potentially useful information.
+  if (summaries) {
+    Object.keys(summaries).forEach((key) => {
+      // We don't want values that are too long, or invalid values
+      const invalidValues = ["null", "undefined", "nan", "", "none"];
+      if (
+        summaries[key].toString().length < 14 &&
+        !invalidValues.includes(summaries[key].toString().toLowerCase()) &&
+        key.toString().length < 10
+      ) {
+        dataPoints.push({
+          icon: TbAxisX,
+          alt: "Cloud Coverage Icon",
+          text: `${beautifyKey(key)} - ${summaries[key].toString()}`,
+          tooltip: key,
+        });
+      }
+    });
+  }
+
+  // If more than 4 data points were found, return only the first 4
+  if (dataPoints.length > 4) {
+    return dataPoints.slice(0, 4);
+  }
+
+  return dataPoints;
+};
+
+export const parseFeatureDataPoints = (feature: Feature): DataPoint[] => {
+  // just return the time
+  const dataPoints: DataPoint[] = [];
+  const { properties } = feature;
+  const { datetime } = properties;
+
+  if (datetime) {
+    dataPoints.push({
+      icon: IoTimeOutline,
+      alt: "Time Icon",
+      text: parseDate(datetime, true),
+      tooltip: "Datetime",
+    });
+  }
+
+  return dataPoints;
+};
+
+export const returnFeatureThumbnail = (feature: Feature): string => {
+  return (
+    feature.links.find((link) => link.rel === "thumbnail")?.href ||
+    "https://via.placeholder.com/110"
+  );
```

Comment:
I'm sure I saw this code in a file further up, or at least very similar, so maybe this function should be extended, if easy to do, to include the `rel` passed in? What is your opinion?

Reply:
There is another case of this, but it's slightly different.

[Its used on the `onError` callback of an `<img/>`](https://github.com/EO-DataHub/eodhp-resource-catalogue-ui/blob/eodhp-578-aoi-item-search/src/pages/MapViewer/components/Toolbox/components/ToolboxRow/index.tsx#L18)

As we don't host the images, sometimes a thumbnail link provided may be behind a paywall or unavailable in some way. Should we still try to figure out a way to merge the functions?